compiler/passes/checkResolved.cpp

```diff
@@ -60,15 +60,26 @@ checkConstLoops() {
   }
 }
 
-static void checkForClassAssignOps(FnSymbol* fn) {
+static void checkForIllegalClassOps(FnSymbol* fn) {
   if (fn->getModule()->modTag == MOD_USER) {
-    if (strcmp(fn->name, "=") == 0 &&
-        fn->formals.head) {
+    if (fn->formals.head && fn->formals.head->next &&
```

Comment:
Shouldn't `fn->formals.size() == 2` be enough?

Reply:
`.length` it seems, but yes.